pg_metadata/dock.py

```diff
@@ -96,6 +167,7 @@ def layer_changed(self, layer):
         # The metadata HTML is wrong if there are many pgmetadata in different databases
 
         for connection_name in connections:
+            self.save_button.setMenu(None)
```

Comment:
1. If you already have metadata displayed in the dock, what happen if you click next on a raster layer ?
It's wrong behavior obviously.

2. This would still let the button active right ? The user will consider it as a bug if he can click on a button but nothing happens. Look what is done in QGIS, according to layer to have activated in the legend, you can click **or** not on a button. You can't click on all buttons.
![Screenshot from 2020-11-25 09-21-34](https://user-images.githubusercontent.com/1609292/100202147-37a58800-2f01-11eb-8773-e19aec7fffb6.png)